src/main/java/org/needle4j/postconstruct/PostConstructProcessor.java

```diff
@@ -29,9 +31,12 @@
      * Internal Container of all Annotations that trigger invocation.
      */
     private final Set<Class<? extends Annotation>> postConstructAnnotations = new HashSet<Class<? extends Annotation>>();
+    private final NeedleConfiguration needleConfiguration;
 
     @SuppressWarnings("unchecked")
-    public PostConstructProcessor(final Set<Class<?>> postConstructAnnotations) {
+    public PostConstructProcessor(final Set<Class<?>> postConstructAnnotations,
```

Comment:
Since we only need the value of strategy here, why passing the complete configuration? I would prefer having a constructor that uses the "DEFAULT" default and another that allows overwriting it.